### PR TITLE
Gasmaks tint and tint calculation update (Oranges approved edition)

### PR DESCRIPTION
**Pull request by Ghilker**
## About The Pull Request
Gasmaks apply a darkening overlay similar to the welding one but much lighter and open. 
Without mask
nomask
With mask
covid
With mask and sunglasses
image

Note that other items that apply tint haven't been changed if worn on their own (sunglasses don't darken the vision) but will increase the darkening effect if staked (mask + sunglasses will give an even darker overlay). 

Oranges' approval
dad2

## Why It's Good For The Game
All upside and no downside are no fun.

## Changelog

:cl:
balance: gasmasks give a slightly darkening overlay, effect will stack with other items that have tint.
code: tint code has been updated to properly work.
/:cl:

### CODE REVIEW

File: code/__DEFINES/inventory.dm
Comment (Ghommie) [inline code/__DEFINES/inventory.dm:137]: Perhaps you should call this `TINT_MINIMAL` hecause it's not zero.

**Comment by drIcepop** [comment]
Uh tider nerf?

File: code/__DEFINES/inventory.dm
Comment (Ghilker) [reply code/__DEFINES/inventory.dm:137]: yer right

**Comment by SuperNovaa41** [comment]
This is a way better tint that the original

**Comment by Ghilker** [comment]
> This is a way better tint that the original, but can you show sunglasses + gas mask

Sunglasses + mask
image

**Comment by SkeletalElite** [comment]
is there even a compelling reason to use a gas mask over a breath other than for hiding your face?

**Comment by Ghilker** [comment]
it hides your face and you get pepper spray protection

**Comment by Burgermancoding** [comment]
Does this change affect miner gas masks? curious because lavaland is already pretty dark and all that

**Comment by Ghilker** [comment]
Gave mining mask a 0.25 tint and same to the ninja mask

**Comment by Sealed101** [comment]
why would a ninja have to have this tint

**Comment by Ghilker** [comment]
0.25 doesn't give you an overlay

**Comment by TheBonded** [comment]
What happens if I alt click the sunglasses to turn on their built in overlay as well, on top of the gas mask? You can alt click most goggles to give your screen a coloured tint

**Comment by Ghilker** [comment]
yes you'll have an overlay on top of the one you receive for the items you are wearing

**Comment by CRITAWAKETS** [comment]
Why is this a thing? It's not really strong or anything, plus the tint isnt even green which would actually be a balance implication (less color vision on gas masks)

**Comment by Ghommie** [comment]
> What happens if I alt click the sunglasses to turn on their built in overlay as well, on top of the gas mask? You can alt click most goggles to give your screen a coloured tint

These glasses color overlays are actually client screen colors, they are fairly different from tint and do not conflict with it.

**Comment by SuperNovaa41** [comment]
> Why is this a thing? It's not really strong of an effect or anything, so nothing's really like, obscured. Plus the tint isnt even green which would actually be a balance implication (less color vision on gas masks)

It's just annoying enough to discourage the people who are constantly wearing them from doing so

**Comment by moocowswag** [comment]
I really don`t think darkening the area around the player and having a spotlight follow them around as if they were the center of attention would be so discouraging, youre just adding another upside to the gas mask.

File: code/modules/clothing/masks/gasmask.dm
Comment (Ghommie) [inline code/modules/clothing/masks/gasmask.dm:156]: Shouldn't syndie masks have a reduced tint value, being (mostly) exclusive to nukies and tacticool?

File: code/_onclick/hud/fullscreen.dm
Comment (Ghommie) [inline code/_onclick/hud/fullscreen.dm:3]: Is there a problem if we just assign a default arg value?
```suggestion
/mob/proc/overlay_fullscreen(category, type, severity, overlay_alpha = 255)
```

File: code/modules/clothing/masks/gasmask.dm
Comment (Ghilker) [reply code/modules/clothing/masks/gasmask.dm:156]: you're right, i'll lower it to 0.25 so that effect doesn't apply

File: code/_onclick/hud/fullscreen.dm
Comment (Ghilker) [reply code/_onclick/hud/fullscreen.dm:3]: no problem, i'll change it

**Comment by Pickle-Coding** [comment]
Why is there a spotlight around the player?

**Comment by Coastfront** [comment]
Ah yes because a mask that can be used for internals, conceals your face and provides pepperspray protection is worthy of getting a moderatly annoying nerf. Not like theres a literal fuckton of other ways to get the exact same effects roundstart.
I just dont see how this is good for the game, at all. Gasmasks arent even that good and people will just change gasmasks from being the thing tiders wear for its benefits to eva helmets+internals. Or bandanas with goggles since iirc those block pepperspray. Or some of the costume helmets that hide your face and dont eat a mask slot. 
Oh right and the extremely minor buff gasmasks got that gives them the extremely shitty filter power. That is just not good justification for this if you even tried to use it as that. Internals protect you from harmful shit in the air infinitely more effectively. Internals also iirc can be used for smoke which gasmasks have blocked for a long time, but even then thats not a super strong or unique ability.
and if this is for the sake of realism, just why? realism isn't always a good thing.  If we wanted to go the realism route for anything why half ass it either. If we're gonna make gasmasks make peoples vision shitty, make literally every mask that obscurs your eyes give your vision the annoying tint.
This is just a lame pr.

**Review by Twaticus — APPROVED** [review]

**Comment by Laz0rG** [comment]
Please give us a thought out reason as to why this makes the game better for everyone playing it besides arbitrary and meaningless statements like "All upside and no downside are no fun."

**Comment by Laz0rG** [comment]
Given how easy it is to acquire the benefits you listed at round start outside of gasmasks (using bandanas and an internal tank respectively) it seems rather pointless to apply this nerf to the gasmask, as the gasmask only provides the advantage of saving a single space of inventory storage compared to the solutions I listed in brackets.

**Comment by SuperNovaa41** [comment]
> Given how easy it is to acquire the benefits you listed at round start outside of gasmasks (using bandanas and an internal tank respectively) it seems rather pointless to apply this nerf to the gasmask, as the gasmask only provides the advantage of saving a single space of inventory storage compared to the solutions I listed in brackets.

Bandanas (and internals masks) are only protection from pepper spray on the mouth. Your eyes are still extremely susceptible. Getting sunglasses is far more difficult in comparison to a bandana.

**Comment by Coastfront** [comment]
> > Given how easy it is to acquire the benefits you listed at round start outside of gasmasks (using bandanas and an internal tank respectively) it seems rather pointless to apply this nerf to the gasmask, as the gasmask only provides the advantage of saving a single space of inventory storage compared to the solutions I listed in brackets.
> 
> Bandanas (and internals masks) are only protection from pepper spray on the mouth. Your eyes are still extremely susceptible. Getting sunglasses is far more difficult in comparison to a bandana.

You could also just put on any type of goggle, since iirc those also provide pepperspray protection to your eyes. And goggles are extremely easily found. Or an eva suit helmet which you can get for free by saying "ai door" outside of eva.

**Comment by Addust** [comment]
i do not approve

**Comment by Addust** [comment]
a riot has started in terry's OOC chat
they demand this be removed
i think you dun goofed up

**Comment by theta006** [comment]
why the fuck is this a thing? the only thing you get from having a gas mask on is you are pepper spray proof no good sec uses that shit and with and you can take the name of a ID you have but talking still shows your name it's not a "meta"

**Comment by optimumtact** [comment]
there is a reason greytiders are pictured as wearing a greyjumpsuit and a gas mask theta

**Comment by theta006** [comment]
and? who cares if a tider has drip

**Comment by theta006** [comment]
now its going to be greysuit and space helmet

**Comment by antrophoss** [comment]
remove this shit

**Comment by optimumtact** [comment]
> 

me obviously

**Comment by theta006** [comment]
> The reason is that gasmasks make your face unknown, protect you from pepper spray and allow you to attach a tank to breathe and negate 90% of atmos threats. So I add a small downside to discourage people from wearing them all the time.

dont you know the box has a mask that can "negate 90% of atmos threats"

**Comment by SinfulBliss** [comment]
Would it be possible to remove the darkening effect from sechuds? Sec requires sechuds to do their job whatsoever, so they don't really have a choice about wearing them.

**Comment by optimumtact** [comment]
or you can take the huds off while wearing the mask in internals environments

**Comment by SinfulBliss** [comment]
> or you can take the huds off while wearing the mask in internals environments

The HUDs are necessary for the job since they show who's wanted. Without them you won't really know who you're chasing or who to arrest. Sec should still be able to do their job in the absence of a perfect atmosphere.

**Comment by optimumtact** [comment]
you can just remember, or ask over sec radio.

please dont' act like huds are a requirement of the job.

**Comment by SinfulBliss** [comment]
> you can just remember, or ask over sec radio.
> 
> please dont' act like huds are a requirement of the job.

That would require examining every person that flashes on your screen to see who matches the name of one of the many wanted people, which is essentially impossible.

**Comment by Hafahya** [comment]
Of fucking course cooders nerf one of symbols of the game, because on their manuel server its 'muh not realistic'. Counters atmos treats? Only atmos treat worth accounting is a plasmaflood, all other happen because of incompetency. Pepper spray is a mass insta stun with no downsides. You nerfed the item because many people were wearing it? Why is that a reason for nerf? It should be the opposite, people love this item. Also, in the year 25XX gas masks not blocking your view is the least non-realistic thing. 
>it should lead to better gameplay because there will be people making the decision in one or the other directions.
How exactly punishing people for making a choice will make gameplay better? There was a lotof people not wearing the mask before, now will be more, its stupid.
There is no good reason for this PR, cooder's excuses are invented on the go and are really shitty, only reason for this PR is how cooders dont like tiders (people that have fun)

**Comment by Laz0rG** [comment]
> Of fucking course cooders nerf one of symbols of the game, because on their manuel server its 'muh not realistic'. Counters atmos treats? Only atmos treat worth accounting is a plasmaflood, all other happen because of incompetency. Pepper spray is a mass insta stun with no downsides. You nerfed the item because many people were wearing it? Why is that a reason for nerf? It should be the opposite, people love this item. Also, in the year 25XX gas masks not blocking your view is the least non-realistic thing.

There really is no justification for this nerf besides an attack on greytide culture, given that gasmasks aren't a  powercrept item. The certain coders responsible for this culture war may as well petition MSO to make all the servers MRP at this point, instead of ruining functionality of the game like they did with the ID console, which was done all in the name is gimping greytiding.

**Comment by ChocolateBickie** [comment]
> Of fucking course cooders nerf one of symbols of the game

This hurts to read.

I don't think I've ever heard a complaint about gas masks being too strong. If I am searching for criminals as sec and see someone with a gas mask, I stun and search them.

I hope Ghilker and Orange unmerge because this...is just bad

**Comment by Coastfront** [comment]
Yeah this is probably the most fucking pointless pr thats happened in a long while and gotten merged. Nerf an item because you dont like it in specific and then make up stupid excuses "oh it negates atmos threats" yeah stop trying to bullshit the players and just be honest. You dont like the gasmask so you dont want people to wear it. If that wasnt your only reason you'd be nerfing all the other items with the same availability that do the same thing.

What a fucking joke.

**Comment by SinfulBliss** [comment]
You gain: Slightly more realism
You lose: Viability of all gasmasks, iconic assistant look, player enjoyment

Gasmasks do not require balancing since internals masks accomplish the exact same goal without the vision loss. Realism also is a bad reason for any code change.

**Comment by theta006** [comment]
> you can just remember, or ask over sec radio.
> 
> please dont' act like huds are a requirement of the job.

ahh yes let's just stand here open to get shot with no sec hud just so I can ask if some tider is wanted and have no one say anything

**Comment by Nebulacrity** [comment]
Yeah, this is a cringe change. I suggest it being reverted. Realism isn't always king, after all.

**Comment by TheBonded** [comment]
there is literally a fix pr already open and people are STILL complaining about it

**Comment by Hafahya** [comment]
> there is literally a fix pr already open and people are STILL complaining about it

We are not complaing about too much tint, we are complaining about it being there at all. This nerf is stupid in of itself

**Comment by optimumtact** [comment]
handsclaspedglasses
get real

**Comment by squi1-rasb** [comment]
woah, coders ignoring the *entire fucking playerbase?* who couldve seen that coming

**Comment by BrandonSmith6** [comment]
_~80% dislike percentage_

**Ight boys, it's time to merge this. WE HAVE UNANIMOUS APPROVAL **

**Comment by TheBonded** [comment]
Does this suck? yeah. is complaining here doing anything? no. if you hate it this much, make a revert pr, or rework it.

**Comment by optimumtact** [comment]
it's helpful if it makes them feel better.

**Comment by Nebulacrity** [comment]
> it's helpful if it makes them feel better.

That's the problem, it makes them feel worse.

**Comment by drIcepop** [comment]
> Does this suck? yeah. is complaining here doing anything? no. if you hate it this much, make a revert pr, or rework it.

I have a slight feeling that a revert PR will get closed instantly much like the singulo/tesla removal revert PR.

**Comment by TheBonded** [comment]
> > Does this suck? yeah. is complaining here doing anything? no. if you hate it this much, make a revert pr, or rework it.
> 
> I have a slight feeling that a revert PR will get closed instantly much like the singulo/tesla removal revert PR.

@drIcepop Then create some sort of alternative. For example, make a gas mask muffle your voice, making you hard to hear by anyone not standing next to you. You could copy paste the whisper code, which already acts this way, and have that effect applied if you are wearing a gas mask. That way it still has a downside, which is what this pr wanted, but the downside is more situational. Make sure not to have the effect apply if the person has the tongue tied quirk, though.

Also, I actually really do like this idea as an alternative downside, so if I get @optimumtact or some other maintainer to approve it during the feature freeze, I'd be happy to begin work on it right away.

**Comment by Nebulacrity** [comment]
> > > Does this suck? yeah. is complaining here doing anything? no. if you hate it this much, make a revert pr, or rework it.
> > 
> > 
> > I have a slight feeling that a revert PR will get closed instantly much like the singulo/tesla removal revert PR.
> 
> @drIcepop Then create some sort of alternative. For example, make a gas mask muffle your voice, making you hard to hear by anyone not standing next to you. You could copy paste the whisper code, which already acts this way, and have that effect applied if you are wearing a gas mask. That way it still has a downside, which is what this pr wanted, but the downside is more situational. Make sure not to have the effect apply if the person has the tongue tied quirk, though.
> 
> Also, I actually really do like this idea as an alternative downside, so if I get @optimumtact or some other maintainer to approve it during the feature freeze, I'd be happy to begin work on it right away.

I'd much rather have this muffled speech thing over the tinted vision. +1

**Comment by Coastfront** [comment]
> > > Does this suck? yeah. is complaining here doing anything? no. if you hate it this much, make a revert pr, or rework it.
> > 
> > 
> > I have a slight feeling that a revert PR will get closed instantly much like the singulo/tesla removal revert PR.
> 
> @drIcepop Then create some sort of alternative. For example, make a gas mask muffle your voice, making you hard to hear by anyone not standing next to you. You could copy paste the whisper code, which already acts this way, and have that effect applied if you are wearing a gas mask. That way it still has a downside, which is what this pr wanted, but the downside is more situational. Make sure not to have the effect apply if the person has the tongue tied quirk, though.
> 
> Also, I actually really do like this idea as an alternative downside, so if I get @optimumtact or some other maintainer to approve it during the feature freeze, I'd be happy to begin work on it right away.

Or, get this, maybe, its not just that this is a completely pointless unfounded nerf, But the fact that we even have to consider "we need to make an arbitrary dumb nerf to something because 3 fucking people ever said x change is good so now it exists"
Yeah complaining won't change it, because if coders fucking listened to community feedback ever, a lot of stuff wouldn't be the way it is, and people wouldn't have a burning hatred of coders.

We shouldn't need to fucking compromise on a nerf, that nobody fucking wants to exist.

**Comment by TheBonded** [comment]
> 



> > > > Does this suck? yeah. is complaining here doing anything? no. if you hate it this much, make a revert pr, or rework it.
> > > 
> > > 
> > > I have a slight feeling that a revert PR will get closed instantly much like the singulo/tesla removal revert PR.
> > 
> > 
> > @drIcepop Then create some sort of alternative. For example, make a gas mask muffle your voice, making you hard to hear by anyone not standing next to you. You could copy paste the whisper code, which already acts this way, and have that effect applied if you are wearing a gas mask. That way it still has a downside, which is what this pr wanted, but the downside is more situational. Make sure not to have the effect apply if the person has the tongue tied quirk, though.
> > Also, I actually really do like this idea as an alternative downside, so if I get @optimumtact or some other maintainer to approve it during the feature freeze, I'd be happy to begin work on it right away.
> 
> Or, get this, maybe, its not just that this is a completely pointless unfounded nerf, But the fact that we even have to consider "we need to make an arbitrary dumb nerf to something because 3 fucking people ever said x change is good so now it exists" Yeah complaining won't change it, because if coders fucking listened to community feedback ever, a lot of stuff wouldn't be the way it is, and people wouldn't have a burning hatred of coders.
> 
> We shouldn't need to fucking compromise on a nerf, that nobody fucking wants to exist.

Well, I don't control what the maintainers want. no one here does. so your choices are to either live with what they say and try to improve things you disagree with or go play somewhere else. I hear beestation is based entirely on older tg code with cloning and stuff not removed. 

Whether you like it or not, the maintainers are the people who decide which prs get merged. It's up to the playerbase to code alternative prs if they don't like pnes that get merged.

**Pull request merged**
